events/events.go

```diff
@@ -74,13 +74,59 @@ func EmitInitComplete(nodeID types.NodeID) {
 	)
 }
 
+func EmitRegisteredInPoet(url string, challenge []byte, roundID string) {
+	const help = "Registered in PoET."
+	emitUserEvent(
+		help,
+		false,
+		&pb.Event_RegisteredInPoet{
+			RegisteredInPoet: &pb.EventRegisteredInPoet{
+				Url:       url,
+				Challenge: challenge,
```

Comment:
When we're emitting events, is there guarantee that the bytes used in `challenge []byte` will not change?
I stumbled upon situations when this implicit requirement was violated by accident more than once, e.g. due to a reused temporary buffer.
Maybe we could use `slices.Clone()` here?

(not insisting, depends on the usage)

Reply:
As I also noted in the related PR in the `api` repository: I don't think we should emit the `challenge` as part of the event at all. It carries little information usable to an operator of a node or even a developer. Since it is a hash it can also not be used to reconstruct a accidentally deleted registration 🤷